seisbench/models/base.py

```diff
@@ -141,60 +143,257 @@ def download_metadata_callback(metadata_path):
+        cls._cleanup_local_repository()
+
+        # Idea: If details, copy all "latest" configs to a temp directory
 
+        model_path = cls._model_path()
         weights = [
-            x[:-3] for x in seisbench.util.ls_webdav(remote_path) if x.endswith(".pt")
+            cls._parse_weight_filename(x)[0]
+            for x in model_path.iterdir()
+            if cls._parse_weight_filename(x)[0] is not None
         ]
 
-        if details:
-            detail_weights = {}
+        if remote:
+            remote_path = cls._remote_path()
+            weights += [
+                cls._parse_weight_filename(x)[0]
+                for x in seisbench.util.ls_webdav(remote_path)
+                if cls._parse_weight_filename(x)[0] is not None
+            ]
 
-            for weight in weights:
+        # Unique
+        weights = sorted(list(set(weights)))
 
-                def download_metadata_callback(metadata_path):
-                    remote_metadata_path = os.path.join(
-                        cls._remote_path(), f"{weight}.json"
-                    )
-                    try:
-                        util.download_http(
-                            remote_metadata_path, metadata_path, progress_bar=False
-                        )
-                    except ValueError:
-                        # A missing metadata file does not lead to a crash
-                        pass
+        if details:
+            return {
+                name: cls._get_latest_docstring(name, remote=remote) for name in weights
+            }
+        else:
+            return weights
 
-                _, metadata_path = cls._pretrained_path(weight)
+    @classmethod
+    def _get_latest_docstring(cls, name, remote):
+        versions = cls.list_versions(name, remote=remote)
+        version_str = max(versions, key=version.parse)
+
+        _, metadata_path = cls._pretrained_path(name, version_str)
 
-                seisbench.util.callback_if_uncached(
-                    metadata_path, download_metadata_callback
+        if metadata_path.is_file():
+            with open(metadata_path, "r") as f:
+                weights_metadata = json.load(f)
+        else:
+            _, remote_metadata_path = cls._get_remote_names(name, version_str)
+            with tempfile.TemporaryDirectory() as tmpdir:
+                metadata_path = Path(tmpdir) / "metadata.json"
+                util.download_http(
+                    remote_metadata_path, metadata_path, progress_bar=False
                 )
+                with open(metadata_path, "r") as f:
+                    weights_metadata = json.load(f)
 
-                if metadata_path.is_file():
-                    with open(metadata_path, "r") as f:
-                        config = json.load(f)
-                    detail_weights[weight] = config.get("docstring", None)
-                else:
-                    detail_weights[weight] = None
+        return weights_metadata.get("docstring", None)
```

Comment:
Similar to the ```from_pretrained``` error. We see the following when trying to call the internal ```_get_latest_docstring``` method:

```python
MissingSchema                             Traceback (most recent call last)
<ipython-input-33-1d1dafe38e58> in <module>
----> 1 sbm.PhaseNet._get_latest_docstring('stead', True)

~/seisbench-production/seisbench/seisbench/models/base.py in _get_latest_docstring(cls, name, remote)
    291             with tempfile.TemporaryDirectory() as tmpdir:
    292                 metadata_path = Path(tmpdir) / "metadata.json"
--> 293                 util.download_http(
    294                     remote_metadata_path, metadata_path, progress_bar=False
    295                 )

~/seisbench-production/seisbench/seisbench/util/file.py in download_http(url, target, progress_bar, desc, precheck_timeout)
     29     seisbench.logger.info(f"Downloading file from {url} to {target}")
     30 
---> 31     precheck_url(url, timeout=precheck_timeout)
     32 
     33     req = requests.get(url, stream=True, headers={"User-Agent": "SeisBench"})

~/seisbench-production/seisbench/seisbench/util/file.py in precheck_url(url, timeout)
     75 
     76     try:
---> 77         req = requests.head(url, timeout=timeout, headers={"User-Agent": "SeisBench"})
     78 
     79         if req.status_code >= 400:  # Assumes all 200 and 300 codes are acceptable

/work/software/conda/2020.11/lib/python3.8/site-packages/requests/api.py in head(url, **kwargs)
    102 
    103     kwargs.setdefault('allow_redirects', False)
--> 104     return request('head', url, **kwargs)
    105 
    106 

/work/software/conda/2020.11/lib/python3.8/site-packages/requests/api.py in request(method, url, **kwargs)
     59     # cases, and look like a memory leak in others.
     60     with sessions.Session() as session:
---> 61         return session.request(method=method, url=url, **kwargs)
     62 
     63 

/work/software/conda/2020.11/lib/python3.8/site-packages/requests/sessions.py in request(self, method, url, params, data, headers, cookies, files, auth, timeout, allow_redirects, proxies, hooks, stream, verify, cert, json)
    514             hooks=hooks,
    515         )
--> 516         prep = self.prepare_request(req)
    517 
    518         proxies = proxies or {}

/work/software/conda/2020.11/lib/python3.8/site-packages/requests/sessions.py in prepare_request(self, request)
    447 
    448         p = PreparedRequest()
--> 449         p.prepare(
    450             method=request.method.upper(),
    451             url=request.url,

/work/software/conda/2020.11/lib/python3.8/site-packages/requests/models.py in prepare(self, method, url, headers, files, data, params, auth, cookies, hooks, json)
    312 
    313         self.prepare_method(method)
--> 314         self.prepare_url(url, params)
    315         self.prepare_headers(headers)
    316         self.prepare_cookies(cookies)

/work/software/conda/2020.11/lib/python3.8/site-packages/requests/models.py in prepare_url(self, url, params)
    386             error = error.format(to_native_string(url, 'utf8'))
    387 
--> 388             raise MissingSchema(error)
    389 
    390         if not host:

MissingSchema: Invalid URL 'stead.pt': No schema supplied. Perhaps you meant http://stead.pt/?
```

Again, perhaps Line #282 is the culprit, as I can't seem to find any version variable in scope.

Reply:
Additional note: This only raises an error if the associated remote files for the pre-trained model have not been downloaded yet and a remote cache lookup is required. If the model (with versioning added) exists in the local cache, it works as expected (which is probably why the associated unit test for the method did not flag this).

Reply:
The culprit is line 290, i.e., instead of generating an URL the code only generated the filename. I fixed this now.